sys/fmt/fmt.c

```diff
@@ -48,19 +48,23 @@ static const uint32_t _tenmap[] = {
 
 #define TENMAP_SIZE  ARRAY_SIZE(_tenmap)
 
-static inline int _is_digit(char c)
+static inline char _to_lower(char c)
 {
-    return (c >= '0' && c <= '9');
+    return 'a' + (c - 'A');
 }
 
-static inline int _is_upper(char c)
+int fmt_is_number(const char *str)
 {
-    return (c >= 'A' && c <= 'Z');
-}
+    if (!str) {
```

Comment:
```suggestion
    if (!str | !*str) {
```

otherwise an empty string would be considered a number.

Reply:
make sense, fixed.